### DISCUSSION TITLE
VideoPlayerDemo

### DISCUSSION THREAD

**Discussion by firemen01** [DelphiWorlds/Kastri#255] [Q&A]
I tried to run the demo project VideoPlayer as is, but it does not run on an android device.
freezes \ closes on FPlayer := TVideoPlayer.Create;
the video player works fine on the same Alcion.
can you tell me what could be the reason?
by the way, in Alcion, different r.jar are substituted for apk and aab.
I am not an expert in this. tell me how to run and try your video player?
customview-poolingcontainer-1.0.0.jar is not connected in the demo, although the description says that it should be.
and by the way, in Alcion, exoplayer 2.19.1, maybe you can update it?
[VideoPlayerDemo.zip](https://github.com/user-attachments/files/17193209/VideoPlayerDemo.zip)

**Comment by firemen01** [comment]
Please also tell me how to manually compile r.jar?  since Codex has too many dependencies.

**Comment by DelphiWorlds** [comment]
> by the way, in Alcion, different r.jar are substituted for apk and aab.

I think you mean Alcinoe, which uses [a deprecated ExoPlayer](https://developer.android.com/media/media3/exoplayer/migration-guide).

> I am not an expert in this. tell me how to run and try your video player?

The demo from Kastri works OK here. I suspect yours does not work due to not deploying the required resources, as per [the demo in Kastri](https://github.com/DelphiWorlds/Kastri/tree/master/Demos/VideoPlayer).

> by the way, in Alcion, exoplayer 2.19.1, maybe you can update it?

As above, Alcinoe's ExoPlayer is actually very much out of date, being version 2. Kastri uses ExoPlayer from the newer Media3 library.

> Please also tell me how to manually compile r.jar? since Codex has too many dependencies.

You do not need to compile Codex in order to use it. The installer for the latest version is [here](https://www.delphiworlds.com/codex/latest). Manually compiling the R.jar takes **a lot of work**, hence why it has been automated in Codex.

In future, please use the [issues](https://github.com/DelphiWorlds/Kastri/issues) to report questions or issues about code in Kastri.
